server/src/features/facilities/facility.controller.js

```diff
@@ -20,6 +25,26 @@ export class FacilitiesController {
     }
   }
 
+  async reserveCourt(req, res, next) {
+    try {
+      if (!req.user) {
+        throw new ApiError(401, "Unauthorized");
+      }
```

Comment:
```suggestion
      }
      const { error } = reserveCourtSchema.validate(req.body);
      if (error) throw new ApiError(400, error.details[0].message);

```
 call the validation function that you will create before going to the service call